samples/bluetooth/fast_pair/locator_tag/src/ui/platform_dk.c

```diff
@@ -138,15 +152,37 @@ static void btn_handle(uint32_t button_state, uint32_t has_changed)
 		app_ui_request_broadcast(APP_UI_REQUEST_FP_ADV_MODE_CHANGE);
 	}
 
-	if (has_changed & button_state & APP_RING_BTN) {
-		app_ui_request_broadcast(APP_UI_REQUEST_RINGING_STOP);
+	if (has_changed & button_state & APP_RING_AND_MOTION_BTN) {
+		/* It is assumed that this function executes in the cooperative thread context. */
+		__ASSERT_NO_MSG(!k_is_preempt_thread());
+		__ASSERT_NO_MSG(!k_is_in_isr());
+
+		if (k_work_delayable_is_pending(&ring_stop_work)) {
```

Comment:
Also, consider what should happen if the user accidentally presses the button 3 or 4 times. I think we shouldn't signal motion and stop ringing requests in case of triple-button press. I think we should log the warning "No action associated with %d button clicks"

Reply:
> You can improve the implementation by relying on the local (static) click counter and the k_uptime_get function instead of using the k_work status. If the uptime difference is too large after the last button press, we can assume that the click action is finished and reset the click counter.

We still need a separate work to be able to stop the ringing in case of single click. I think it's cleaner and simpler to leave it as is. We don't need a perfect precision for this.

Reply:
> Also, consider what should happen if the user accidentally presses the button 3 or 4 times. I think we shouldn't signal motion and stop ringing requests in case of triple-button press. I think we should log the warning "No action associated with %d button clicks"

I think it's not needed. On Windows OS if you click 4 times on the icon it opens the icon twice. If you click 3 times it opens the icon and highlights it in 3rd click.